### ISSUE TITLE
UI settings pages don't count as editorFocus

### ISSUE THREAD

**Issue by ablacklama** [microsoft/vscode#129717]
Issue Type: <b>Bug</b>

I'm trying to setup hotkeys that will focus on the terminal if an editor is focused. The when condition `editorFocus` works the vast majority of the time but it doesn't work when a UI settings page (Settings, Keyboard Shortcuts, ect.) is open. Is there a good reason for not counting settings pages as `editorFocus`? And if so, can we get some documentation on how to reference that special type of focus? 

Thanks a bunch!

VS Code version: Code 1.58.2 (c3f126316369cd610563c75b1b1725e0679adfb3, 2021-07-14T22:10:15.214Z)
OS version: Windows_NT x64 10.0.19042
Restricted Mode: No

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|AMD Ryzen 9 5900X 12-Core Processor             (24 x 3700)|
|GPU Status|2d_canvas: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: enabled<br>opengl: enabled_on<br>rasterization: enabled<br>skia_renderer: enabled_on<br>video_decode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|31.94GB (20.04GB free)|
|Process Argv|--crash-reporter-id 0e2b899c-1a5d-43cb-be1d-1dada59d9714|
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (51)</summary>

Extension|Author (truncated)|Version
---|---|---
better-comments|aar|2.1.0
Bookmarks|ale|13.1.0
aws-toolkit-vscode|ama|1.27.0
google-cloud-platform-ssh|and|1.0.0
vscode-custom-css|be5|6.0.0
file-properties-viewer|bru|0.5.0
vscode-pytest|Cam|0.1.1
vscode-markdownlint|Dav|0.42.1
py-pack|dav|1.2.0
python-preview|don|0.0.4
githistory|don|0.6.17
gitlens|eam|11.6.0
LogFileHighlighter|emi|2.11.0
vscode-great-icons|emm|2.1.79
vscode-fold|fel|1.4.3
cloudcode|goo|1.13.1
gc-excelviewer|Gra|3.0.42
ai-docstring|gra|0.1.3
vscode-test-explorer|hbe|2.20.4
format-python-docstrings|ian|1.2.1
output-colorizer|IBM|0.1.2
python-coding-conventions|igr|0.0.4
vscode-python-test-adapter|lit|0.6.8
remotefs|lix|0.0.16
aws-cli-configure|mar|0.3.0
theme-monokai-pro-vscode|mon|1.1.19
vscode-docker|ms-|1.15.0
python|ms-|2021.7.1060902895
vscode-pylance|ms-|2021.7.5
jupyter|ms-|2021.8.1054968649
remote-containers|ms-|0.187.1
remote-ssh|ms-|0.65.7
remote-ssh-edit|ms-|0.65.7
vscode-remote-extensionpack|ms-|0.21.0
test-adapter-converter|ms-|0.0.13
pytest-fixtures|nic|0.1.10
autodocstring|njp|0.5.4
vetur|oct|0.34.1
python-dependencies-vscode|pat|0.0.18
vscode-auto-scroll|pej|1.0.4
material-icon-theme|PKi|4.8.0
jinjahtml|sam|0.16.0
trailing-spaces|sha|0.3.1
code-spell-checker|str|1.10.2
ayu|tea|0.20.1
vscode-aws-console|vsc|0.0.5
vscode-icons|vsc|11.5.0
gitblame|wad|8.1.0
jinja|who|0.0.8
markdown-all-in-one|yzh|3.4.0
material-theme|zhu|3.11.2

(17 theme extensions excluded)

</details><details>
<summary>A/B Experiments</summary>

```
vsliv368cf:30146710
vsreu685:30147344
python383cf:30185419
pythonvspyt602:30300191
vspor879:30202332
vspor708:30202333
vspor363:30204092
pythonvspyt639:30300192
pythontb:30283811
pythonvspyt551:30311712
vspre833:30321513
pythonptprofiler:30281270
vshan820:30294714
vstes263:30335439
pythondataviewer:30285071
vscus158:30321503
pythonvsuse255:30340121
vscod805:30301674
pythonvspyt200:30340761
vscextlangct:30333562
binariesv615:30325510
vsccppwt:30329788
pythonvssor306:30341831
bridge0708:30335490

```

</details>

**Comment by roblourens** [comment]
Yeah I think it should. You could use `inSettingsEditor` as a workaround

**Comment by ablacklama** [comment]
I just tried `editorFocus || inSettingsEditor` and it still doesn't seem to work.

**Comment by roblourens** [comment]
Works for me, can you share the exact shortcut setup? You can also use the "Inspect Context Keys" command to debug (they show up in the devtools console)

**Comment by ablacklama** [comment]
```json
{
    "key": "ctrl+shift+alt+enter",
    "command": "workbench.action.terminal.focus",
    "when": "editorFocus || inSettingsEditor"
}
```
I'm not sure how to use the inspect context key. I get output fine, i just don't know where to click or what variables to pay attention too.

**Comment by roblourens** [comment]
This works for me. Where is focus when you press the keybinding? That's where you would click

**Comment by ablacklama** [comment]
ooh that makes sense. It works now. Thank you.
